fs/operations/operations.go

```diff
@@ -1315,6 +1315,48 @@ func NeedTransfer(dst, src fs.Object) bool {
 	return true
 }
 
+// UploadHTTPBody upload an object from http to dst
+func UploadHTTPBody(fdst fs.Fs, body io.ReadCloser, contentLength int64, remote string) error {
+	if contentLength >= 0 {
+		// Size known use Put
+		accounting.Stats.Transferring(remote)
+		body := ioutil.NopCloser(body)                                   // we let the server close the body
+		in := accounting.NewAccountSizeName(body, contentLength, remote) // account the transfer (no buffering)
+		var err error
+		defer func() {
+			closeErr := in.Close()
+			if closeErr != nil {
+				fs.Errorf(remote, "Post request: close failed: %v", closeErr)
+				if err == nil {
+					err = closeErr
+				}
+			}
+			ok := err == nil
+			accounting.Stats.DoneTransferring(remote, err == nil)
+			if !ok {
```

Comment:
I think this `if` statement and the `account.Stats.Error` is wrong (it was wrong in the original code too!).

I think we should move it up into the caller so L333 in the restic.go or thereabouts.

You don't need to call it at all for `copyurl` since you are returning an error straight away from your code.